configure

```diff
@@ -7328,23 +7221,7 @@ cat > config.h.new << EOF
 $_config_h_data
 $(cat config.h.engines)
 
-/* Data types */
-#ifndef SCUMMVM_DONT_DEFINE_TYPES
-typedef unsigned $type_1_byte byte;
-typedef unsigned int uint;
-typedef unsigned $type_1_byte uint8;
-typedef unsigned $type_2_byte uint16;
-typedef unsigned $type_4_byte uint32;
-typedef unsigned $type_8_byte uint64;
-typedef signed $type_1_byte int8;
-typedef signed $type_2_byte int16;
-typedef signed $type_4_byte int32;
-typedef signed $type_8_byte int64;
-#endif
-
-typedef $signed_type_ptr intptr;
-typedef $unsigned_type_ptr uintptr;
-
+/* TODO: What does this do? */
 #if defined(__APPLE__) && !defined(__ppc__)
```

Comment:
This dates back to 00242281006dd26204a3b438f1ecb38928bfb75f, which first defined 64-bit types in ScummVM. It's trying to prevent Apple's `_types/_uint64_t.h` from defining `uint64_t`; that file does:

```c
#ifndef _UINT64_T
#define _UINT64_T
typedef unsigned long long uint64_t;
#endif /* _UINT64_T */
```

Reply:
Did some testing on modern macOS (14.6.1), and I don't think this is necessary anymore. With this PR and the `#if defined(__APPLE__) && !defined(__ppc__)` block deleted, everything works just fine without redefined types errors.

Reply:
Thanks for looking into this, @mistydemeo! I'm so happy to have the mighty Misty of Tigerbrew in our team, to know stuff like this! :smiley: 

Maybe something like i386 Tiger or Leopard needed this, but we don't have any official support for it, anyway. I do have a Leopard i386 VM, but no interest in reviving a ScummVM port for it, at the moment (one can just run the PPC binary through Rosetta 1 just fine). 

So I agree with removing this line. If it either breaks the (maintained) macOS 10.6+ build or the non-existing 10.4/10.5 i386 port, we might as well restore this line if it's proven to still be required there.